## プレ実験のデータ分析

### 50曲を対象に4種のプライミング＋プライミング無しを比較

- chord：コード進行
- lyrics：歌詞
- bpm：BPM
- playcount：再生回数
- none：なし

In [1]:
import pandas as pd
pd.options.display.max_rows = None

In [2]:
with open('./result.csv', 'r') as f:
    results = pd.read_csv(f)

### 50曲のプライミングパターンごとの評価値（ユーザの平均をとったもの）

In [3]:
results = results.sort_values(by=['m_id', 'prime'])
results = results.groupby(['m_id', 'prime']).mean()
results

C1   C2   C3   C4   C5   C6  Valence  Arousal  like
m_id prime                                                          
1    bpm       -0.5 -1.5  0.5 -0.5  1.5  0.5     -0.5      0.5  -0.5
     chord     -1.0 -1.0  1.0  0.0  1.0 -1.0     -1.0      0.0   0.0
     lyrics     1.0  0.0  1.0  0.0  0.0  0.0      0.0      1.0   0.0
     none       1.0 -0.5  1.0  1.5 -0.5 -1.5     -0.5      1.5   0.5
     playcount -0.5 -0.5  0.5  0.5  1.0  0.5     -1.0      0.5   0.0
2    bpm        2.0  0.0  1.0  2.0  1.0  2.0      0.0      2.0   1.0
     chord      0.0  0.0 -0.5  1.0 -2.0 -1.0      0.5      0.5   1.0
     lyrics     0.5  1.0 -0.5  1.0  0.0  0.5      0.5      0.5   1.0
     none       0.5  1.5 -1.0  1.0 -2.0  0.0      1.0      1.0   1.5
     playcount  1.0  2.0 -1.0  1.0  0.0  1.0      1.0      1.0   1.0
3    bpm        1.0  1.0 -1.0  1.0  1.0  1.0      1.0     -1.0   0.0
     chord      1.0  2.0 -2.0  0.5  0.5  0.5      1.5      0.0   1.0
     lyrics     1.0  1.5 -0.5  0.5  1.5  0.5      0.0      0.5   0.5
     none       0.5  1.0 -0.5  0.0  0.5  0.5      0.5      0.0   0.5
     playcount  1.0  1.0 -1.0  0.5  0.5 -0.5      0.5      0.0   1.5
4    bpm        0.0  1.0  1.0  1.0 -2.0  1.0      0.0      1.0   2.0
     chord      1.0  1.0  0.0  0.0  0.0  0.0      1.0      0.0   0.0
     lyrics     0.0 -1.0  0.0 -1.0 -0.5  0.0     -1.0     -1.0   1.5
     none       0.0  1.0 -0.5 -1.0  0.0  0.0      0.0      0.0   0.0
     playcount  0.0  0.5  1.0  0.0  0.0  1.0      0.0      0.0   0.5
5    bpm       -1.0  2.0 -2.0 -2.0  1.0  2.0      2.0     -2.0   1.0
     chord     -0.5  0.0  0.0 -1.5  1.0  1.5      0.5     -1.5   0.0
     lyrics    -1.0  0.5 -2.0 -2.0  1.0  2.0      1.0     -1.5   0.5
     none      -0.5  0.5 -0.5 -2.0  1.5  1.0      0.5     -1.0   0.0
     playcount -2.0 -2.0  1.0 -2.0  0.0  1.0     -2.0     -2.0  -2.0
6    bpm        0.5 -1.0  1.5  0.5 -1.5 -2.0     -1.5      0.0  -0.5
     chord     -0.5 -0.5  1.0  1.0 -0.5 -1.0     -0.5      1.0   0.0
     lyrics     1.0  1.0  1.0  0.0  0.0 -1.0     -1.0      0.0   0.0
     none      -1.0 -1.0  0.0  1.5 -1.5 -2.0     -1.0      1.5   0.0
     playcount -1.0 -1.5  1.5  1.0 -1.5 -2.0     -1.0      0.5  -1.0
7    bpm        1.0  0.0  1.0  2.0 -2.0  1.0     -1.0      2.0  -2.0
     chord      1.0 -0.5  0.0  1.0 -1.5 -1.5     -1.0      1.0   0.0
     lyrics     2.0  0.0  0.0  1.0  0.0  0.0     -1.0      1.0   0.0
     none       0.0 -1.0  1.0  2.0 -2.0 -1.0     -1.0      2.0   1.0
     playcount  1.0  1.0 -0.5  1.0 -1.0  0.0      0.0      0.5   1.0
8    bpm        2.0 -2.0  0.0 -0.5 -2.0 -2.0     -0.5      1.5  -0.5
     chord     -0.5 -1.5  1.0  1.0 -1.5 -1.0     -1.0      1.0  -0.5
     lyrics     0.5  0.0  0.5  1.5 -0.5 -1.0      0.0      1.5   0.0
     none       2.0 -1.0  1.0  2.0 -2.0 -1.0     -1.0      2.0  -1.0
     playcount  1.0 -1.0  1.0  0.0  0.0  0.0     -1.0      1.0  -1.0
9    bpm        0.5  1.5 -1.0 -0.5 -2.0  1.0      1.0     -0.5   1.5
     chord      1.0  1.0  0.0 -1.0  0.0  1.0     -1.0     -1.0  -1.0
     lyrics     1.0  1.0  0.0  0.5 -0.5  1.0      1.0      0.0   1.5
     none       1.0  1.0 -1.0  0.0  0.0  2.0      2.0      0.0   1.0
     playcount  0.5  1.0  0.0 -1.5 -2.0  0.5      0.0     -0.5   1.0
10   bpm        1.0  1.0 -1.5  0.5  1.0 -0.5      0.0      0.5   0.0
     chord      1.0 -0.5  0.5  1.0  1.0  0.5     -0.5      0.0   0.0
     lyrics     2.0  1.0 -1.0  0.0  0.0  0.0      1.0      1.0   0.0
     none       0.0 -1.0 -0.5  1.0 -1.0 -1.0     -0.5      0.5  -0.5
     playcount  0.5  0.5  0.0  0.0  0.0 -0.5      0.5      0.0   0.0
11   bpm       -2.0 -1.0  1.5 -2.0 -2.0 -0.5     -1.5     -1.5   0.0
     chord     -1.0  1.0  1.0 -2.0 -2.0  2.0      0.0     -1.0   0.0
     lyrics    -0.5  0.5  1.5  0.0 -0.5  0.0     -0.5     -0.5   0.5
     none      -0.5 -2.0  1.0 -0.5 -1.0 -1.0     -1.5     -0.5  -1.5
     playcount -1.0  1.0  1.0  1.0  0.0  1.0      0.0      0.0   0.0
12   bpm        0.0 -0.5  0.5  0.5 -2.0 -1.0      0.0      1.0   0.0
     chord     

In [4]:
none = results.xs('none', level=1)
chord = results.xs('chord', level=1)
lyrics = results.xs('lyrics', level=1)
bpm = results.xs('bpm', level=1)
playcount = results.xs('playcount', level=1)

In [5]:
dif_chord = chord - none
# dif_chord

In [6]:
dif_lyrics = lyrics - none
# dif_lyrics

In [7]:
dif_bpm = bpm - none
# dif_bpm

In [8]:
dif_playcount = playcount - none
# dif_playcount

### 各プライミングごとに，プライミングなしとの差の（50曲の）平均を求めたもの

In [9]:
output = pd.concat(
    [dif_chord.mean(), dif_lyrics.mean(), dif_bpm.mean(), dif_playcount.mean()],
    axis=1,
    keys=['chord', 'lyrics', 'bpm', 'playcount']
)
output = output.T
output

C1    C2    C3    C4    C5    C6  Valence  Arousal  like
chord     -0.06  0.02  0.15  0.23 -0.10  0.07    -0.02    -0.09  0.08
lyrics     0.16 -0.01  0.12  0.15  0.18  0.08    -0.02    -0.10  0.15
bpm       -0.06  0.02 -0.01  0.12 -0.13  0.02     0.08    -0.15  0.11
playcount  0.12  0.05 -0.08  0.10  0.06  0.04    -0.06    -0.02  0.00

### 結果

- 好感度（like）の中では，lyricsが一番差がでた．ただし平均0.15の差だからかなり低い
- 好感度の平均はプライミングによって下がるものはなかった